lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -189,14 +189,11 @@ def path_to_3d_segment_with_codes(path, zs=0, zdir='z'):
     """Convert a path to a 3D segment with path codes."""
 
     zs = np.broadcast_to(zs, len(path))
-    seg = []
-    codes = []
     pathsegs = path.iter_segments(simplify=False, curves=False)
-    for (((x, y), code), z) in zip(pathsegs, zs):
-        seg.append((x, y, z))
-        codes.append(code)
+    seg_codes = [((x, y, z), code) for ((x, y), code), z in zip(pathsegs, zs)]
+    seg, codes = zip(*seg_codes)
```

Comment:
This should be protected by an empty-check, because it'll fail if `pathsegs` or `zs` is empty. With the previous implementation, you got back empty lists.